File: Ent_UsAdmin/AgregarEmpresa.vb
Imports System.ComponentModel
Imports System.Data.OleDb

Public Class AgregarEmpresa

    Dim idEmpresaSeleccionada As String
    Public bdEmpresaSeleccionada As String
    Public ipEmpresaSeleccionada As String
    Public nombreEmpresaSeleccionada As String
    Dim ncargando As Cargando
    Dim heightFlow As Integer
    Dim grande As Boolean




    Private Sub ModificarEmpresa_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        FlowUsuarios.Height = 0

        ncargando = New Cargando
        ncargando.MonoFlat_Label1.Text = "Cargando Información"
        ncargando.Show()
        BackgroundInformacionEmpresa.RunWorkerAsync()


    End Sub

    Private Sub BackgroundActualizar_DoWork(sender As Object, e As DoWorkEventArgs) Handles BackgroundActualizar.DoWork
        iniciarconexion()
        Dim idEmpresaGenerado As Integer
        Dim comandoActualizar As OleDbCommand
        Dim consultaActualizar As String
        consultaActualizar = "insert into empresas values('" & txtNombre.Text & "','" & txtRazonSocial.Text & "','" & txtRFC.Text & "','" & txtCalle.Text & "','" & txtNoExterior.Text & "','" & txtColonia.Text & "','" & txtCiudad.Text & "','" & txtEstado.Text & "','" & txtCP.Text & "','" & txtTelefono.Text & "',
                               '" & txtIP.Text & "','" & txtBD.Text & "','" & txtPrefijo.Text & "','" & txtCorreo.Text & "','" & txtPassword.Text & "') select SCOPE_IDENTITY()"
        comandoActualizar = New OleDbCommand
        comandoActualizar.Connection = conexion
        comandoActualizar.CommandText = consultaActualizar
        idEmpresaGenerado = comandoActualizar.ExecuteScalar


        Dim nCargar As Integer = 0
        For Each ctrlc As Control In FlowUsuarios.Controls
            If TypeOf (ctrlc) Is CheckBox Then
                Dim checkE As CheckBox = ctrlc
                If checkE.CheckState = CheckState.Checked Then
                    nCargar += 1

                End If
            End If
        Next

        If nCargar > 0 Then
            For Each ctrlc As Control In FlowUsuarios.Controls
                If TypeOf (ctrlc) Is CheckBox Then
                    Dim checkE As CheckBox = ctrlc
                    If checkE.CheckState = CheckState.Checked Then
                        Dim comandoEmpresasPermitidas As OleDbCommand
                        Dim consultaEmpresasPermitidas As String
                        consultaEmpresasPermitidas = "insert into empresaspermitidas values('" & idEmpresaGenerado & "','" & checkE.Name & "')"
                        comandoEmpresasPermitidas = New OleDbCommand
                        comandoEmpresasPermitidas.Connection = conexion
                        comandoEmpresasPermitidas.CommandText = consultaEmpresasPermitidas
                        comandoEmpresasPermitidas.ExecuteNonQuery()


                    End If
                End If
            Next
        End If
    End Sub

    Private Sub BackgroundActualizar_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundActualizar.RunWorkerCompleted
        Empresas.Actualizado = True


        ncargando.Close()
    End Sub

    Private Sub BunifuThinButton21_Click(sender As Object, e As EventArgs) Handles BunifuThinButton21.Click
        ncargando = New Cargando
        ncargando.MonoFlat_Label1.Text = "Agregando Empresa"
        ncargando.Show()
        BackgroundActualizar.RunWorkerAsync()

    End Sub

    Private Sub Panel1_Paint(sender As Object, e As PaintEventArgs) Handles Panel1.Paint

    End Sub

    Private Sub Panel1_MouseDown(sender As Object, e As MouseEventArgs) Handles Panel1.MouseDown
        If e.Button = MouseButtons.Left Then
            MoveForm(Me)
        End If
    End Sub

    Private Sub ModificarEmpresa_FormClosed(sender As Object, e As FormClosedEventArgs) Handles Me.FormClosed
        Me.Invoke(Sub()
                      Empresas.CargarEmpresas()
                  End Sub)

    End Sub

    Private Sub BackgroundInformacionEmpresa_DoWork(sender As Object, e As DoWorkEventArgs) Handles BackgroundInformacionEmpresa.DoWork
        iniciarconexion()
        Dim comandoUsuarios As OleDbCommand
        Dim consultaUsuarios As String
        Dim readerUsuarios As OleDbDataReader

        consultaUsuarios = "select idusr,nm,nm_complete from usr"
        comandoUsuarios = New OleDbCommand
        comandoUsuarios.Connection = conexion
        comandoUsuarios.CommandText = consultaUsuarios
        readerUsuarios = comandoUsuarios.ExecuteReader
        If readerUsuarios.HasRows Then
            While readerUsuarios.Read
                Dim CheckUsuarios As New CheckBox

                CheckUsuarios.Visible = True
                '  checkEmpresa.Location = New Point(botonempresa.Width - checkEmpresa.Width, botonempresa.Height - checkEmpresa.Height)
                CheckUsuarios.BringToFront()
                '  checkEmpresa.Parent = botonempresa
                CheckUsuarios.ForeColor = Color.White
                CheckUsuarios.Name = readerUsuarios("idusr")
                CheckUsuarios.Tag = readerUsuarios("nm")
                CheckUsuarios.Text = readerUsuarios("nm_complete")
                CheckUsuarios.AutoSize = True
                Me.Invoke(Sub()
                              FlowUsuarios.Controls.Add(CheckUsuarios)
                          End Sub)

            End While
        End If
    End Sub

    Private Sub TimerGrande_Tick(sender As Object, e As EventArgs) Handles TimerGrande.Tick
        If heightFlow <= 272 Then
            FlowUsuarios.Height = heightFlow
            heightFlow += 10
        Else
            TimerGrande.Stop()
            TimerGrande.Enabled = False

        End If
    End Sub

    Private Sub TimerChico_Tick(sender As Object, e As EventArgs) Handles TimerChico.Tick
        If heightFlow >= 0 Then
            FlowUsuarios.Height = heightFlow
            heightFlow -= 10
        Else
            TimerChico.Stop()
            TimerChico.Enabled = False

        End If
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        If grande = False Then
            grande = True
            TimerGrande.Enabled = True
            TimerGrande.Interval = 10
            TimerGrande.Start()

        Else
            grande = False
            TimerChico.Enabled = True
            TimerChico.Interval = 10
            TimerChico.Start()

        End If
    End Sub

    Private Sub BackgroundInformacionEmpresa_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundInformacionEmpresa.RunWorkerCompleted
        ncargando.Close()
    End Sub
End Class

File: Ent_UsAdmin/Configuracion/AgregarEmpresa.cs
using System;
using System.ComponentModel;
using System.Data.OleDb;
using System.Drawing;
using System.Windows.Forms;
using Microsoft.VisualBasic.CompilerServices;

namespace ConfiaAdmin
{

    public partial class AgregarEmpresa
    {

        private string idEmpresaSeleccionada;
        public string bdEmpresaSeleccionada;
        public string ipEmpresaSeleccionada;
        public string nombreEmpresaSeleccionada;
        private Cargando ncargando;
        private int heightFlow;
        private bool grande;

        public AgregarEmpresa()
        {
            InitializeComponent();
        }




        private void ModificarEmpresa_Load(object sender, EventArgs e)
        {
            FlowUsuarios.Height = 0;

            ncargando = new Cargando();
            ncargando.MonoFlat_Label1.Text = "Cargando Información";
            ncargando.Show();
            BackgroundInformacionEmpresa.RunWorkerAsync();


        }

        private void BackgroundActualizar_DoWork(object sender, DoWorkEventArgs e)
        {
            Module1.iniciarconexion();
            int idEmpresaGenerado;
            OleDbCommand comandoActualizar;
            string consultaActualizar;
            consultaActualizar = "insert into empresas values('" + txtNombre.Text + "','" + txtRazonSocial.Text + "','" + txtRFC.Text + "','" + txtCalle.Text + "','" + txtNoExterior.Text + "','" + txtColonia.Text + "','" + txtCiudad.Text + "','" + txtEstado.Text + "','" + txtCP.Text + "','" + txtTelefono.Text + @"',
                               '" + txtIP.Text + "','" + txtBD.Text + "','" + txtPrefijo.Text + "','" + txtCorreo.Text + "','" + txtPassword.Text + "') select SCOPE_IDENTITY()";
            comandoActualizar = new OleDbCommand();
            comandoActualizar.Connection = Module1.conexion;
            comandoActualizar.CommandText = consultaActualizar;
            idEmpresaGenerado = Conversions.ToInteger(comandoActualizar.ExecuteScalar());


            int nCargar = 0;
            foreach (Control ctrlc in FlowUsuarios.Controls)
            {
                if (ctrlc is CheckBox)
                {
                    CheckBox checkE = (CheckBox)ctrlc;
                    if (checkE.CheckState == CheckState.Checked)
                    {
                        nCargar += 1;

                    }
                }
            }

            if (nCargar > 0)
            {
                foreach (Control ctrlc in FlowUsuarios.Controls)
                {
                    if (ctrlc is CheckBox)
                    {
                        CheckBox checkE = (CheckBox)ctrlc;
                        if (checkE.CheckState == CheckState.Checked)
                        {
                            OleDbCommand comandoEmpresasPermitidas;
                            string consultaEmpresasPermitidas;
                            consultaEmpresasPermitidas = "insert into empresaspermitidas values('" + idEmpresaGenerado + "','" + checkE.Name + "')";
                            comandoEmpresasPermitidas = new OleDbCommand();
                            comandoEmpresasPermitidas.Connection = Module1.conexion;
                            comandoEmpresasPermitidas.CommandText = consultaEmpresasPermitidas;
                            comandoEmpresasPermitidas.ExecuteNonQuery();


                        }
                    }
                }
            }
        }

        private void BackgroundActualizar_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            My.MyProject.Forms.Empresas.Actualizado = true;


            ncargando.Close();
        }

        private void BunifuThinButton21_Click(object sender, EventArgs e)
        {
            ncargando = new Cargando();
            ncargando.MonoFlat_Label1.Text = "Agregando Empresa";
            ncargando.Show();
            BackgroundActualizar.RunWorkerAsync();

        }

        private void Panel1_Paint(object sender, PaintEventArgs e)
        {

        }

        private void Panel1_MouseDown(object sender, MouseEventArgs e)
        {
            if (e.Button == MouseButtons.Left)
            {
                Module_resize.MoveForm(this);
            }
        }

        private void ModificarEmpresa_FormClosed(object sender, FormClosedEventArgs e)
        {
            Invoke(new Action(() => My.MyProject.Forms.Empresas.CargarEmpresas()));

        }

        private void BackgroundInformacionEmpresa_DoWork(object sender, DoWorkEventArgs e)
        {
            Module1.iniciarconexion();
            OleDbCommand comandoUsuarios;
            string consultaUsuarios;
            OleDbDataReader readerUsuarios;

            consultaUsuarios = "select idusr,nm,nm_complete from usr";
            comandoUsuarios = new OleDbCommand();
            comandoUsuarios.Connection = Module1.conexion;
            comandoUsuarios.CommandText = consultaUsuarios;
            readerUsuarios = comandoUsuarios.ExecuteReader();
            if (readerUsuarios.HasRows)
            {
                while (readerUsuarios.Read())
                {
                    var CheckUsuarios = new CheckBox();

                    CheckUsuarios.Visible = true;
                    // checkEmpresa.Location = New Point(botonempresa.Width - checkEmpresa.Width, botonempresa.Height - checkEmpresa.Height)
                    CheckUsuarios.BringToFront();
                    // checkEmpresa.Parent = botonempresa
                    CheckUsuarios.ForeColor = Color.White;
                    CheckUsuarios.Name = Conversions.ToString(readerUsuarios["idusr"]);
                    CheckUsuarios.Tag = readerUsuarios["nm"];
                    CheckUsuarios.Text = Conversions.ToString(readerUsuarios["nm_complete"]);
                    CheckUsuarios.AutoSize = true;
                    Invoke(new Action(() => FlowUsuarios.Controls.Add(CheckUsuarios)));

                }
            }
        }

        private void TimerGrande_Tick(object sender, EventArgs e)
        {
            if (heightFlow <= 272)
            {
                FlowUsuarios.Height = heightFlow;
                heightFlow += 10;
            }
            else
            {
                TimerGrande.Stop();
                TimerGrande.Enabled = false;

            }
        }

        private void TimerChico_Tick(object sender, EventArgs e)
        {
            if (heightFlow >= 0)
            {
                FlowUsuarios.Height = heightFlow;
                heightFlow -= 10;
            }
            else
            {
                TimerChico.Stop();
                TimerChico.Enabled = false;

            }
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (grande == false)
            {
                grande = true;
                TimerGrande.Enabled = true;
                TimerGrande.Interval = 10;
                TimerGrande.Start();
            }

            else
            {
                grande = false;
                TimerChico.Enabled = true;
                TimerChico.Interval = 10;
                TimerChico.Start();

            }
        }

        private void BackgroundInformacionEmpresa_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            ncargando.Close();
        }
    }
}